pkg/scheduler/objects/queue.go

```diff
@@ -401,6 +401,28 @@ func (sq *Queue) GetQueueInfos() dao.QueueDAOInfo {
 	return queueInfo
 }
 
+func (sq *Queue) GetPartitionQueues() dao.PartitionQueueDAOInfo {
+	queueInfo := dao.PartitionQueueDAOInfo{}
+	if len(sq.children) > 0 {
+		for _, child := range sq.GetCopyOfChildren() {
+			queueInfo.Children = append(queueInfo.Children, child.GetPartitionQueues())
+		}
+	}
+	queueInfo.QueueName = sq.GetQueuePath()
+	queueInfo.Status = sq.stateMachine.Current()
```

Comment:
Is it safe to get this variable w/o readlock ?

Reply:
Added readlock to ensure its safety. Also replaced write lock with read lock in GetQueuePath() which got introduced as part of cache refactoring. Read lock is good enough in this case. Please verify this as well.

Reply:
Looks good

Reply:
looks good